File: RSMK2/RSMK2/MaterialOperationImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RSMK2
{
    public class MaterialOperationImpl : MarshalByRefObject, MaterialOperation
    {
        public static List<Material> lst = new List<Material>();
        public List<Material> GetAll() => lst;
        public List<Material> DeleteAt(int index)
        {
            lst.RemoveAt(index);
            return lst;
        }
        public List<Material> Add(Material newItem)
        {
            lst.Add(newItem);
            return lst;
        }
        public double CountTotal()
        {
            double result = 0d;
            foreach (var item in lst)
            {
                result += item.GetPrice() * item.GetAmount();
            }
            return result;
        }
    }
}

File: zwsserver/src/operation/MaterialOperationImpl.java
package operation;

import types.Material;

import java.util.ArrayList;
import java.util.List;

public class MaterialOperationImpl implements MaterialOperation {

    public static List<Material> lst = new ArrayList<Material>();

    static {
        lst.add(new Material("rocks", "no", 150, 2000));
        lst.add(new Material("soap", "smells amazing", 23, 25));
    }

    @Override
    public List<Material> GetAll()
    {
        return lst;
    }

    @Override
    public List<Material> DeleteAt(int index)
    {
        lst.remove(index);
        return lst;
    }

    @Override
    public List<Material> Add(Material newItem)
    {
        lst.add(newItem);
        return lst;
    }

    @Override
    public double CountTotal()
    {
        double result = 0d;
        for (var item : lst)
        {
            result += item.getPrice() * item.getAmount();
        }
        return result;
    }

}
